hikari/impl/rest.py

```diff
@@ -2786,6 +2786,22 @@ async def edit_guild(
         assert isinstance(response, dict)
         return self._entity_factory.deserialize_rest_guild(response)
 
+    @typing_extensions.override
+    async def set_guild_incident_actions(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        *,
+        invites_disabled_until: undefined.UndefinedNoneOr[datetime.datetime] = undefined.UNDEFINED,
+        dms_disabled_until: undefined.UndefinedNoneOr[datetime.datetime] = undefined.UNDEFINED,
+    ) -> guilds.GuildIncidents:
+        route = routes.PUT_GUILD_INCIDENT_ACTIONS.compile(guild=guild)
+        body = data_binding.JSONObjectBuilder()
+        body.put("invites_disabled_until", invites_disabled_until.isoformat() if invites_disabled_until else None)
+        body.put("dms_disabled_until", dms_disabled_until.isoformat() if dms_disabled_until else None)
+        response = await self._request(route, json=body)
+        assert isinstance(response, dict)
+        return self._entity_factory.deserialize_guild_incidents(response)
```

Comment:
If this is called with `func(dms_disabled_until=...)`, but not `invites_disabled_until`, then we will be sending it as `None`, which would reset any old values. We do not want to change it

```suggestion
        route = routes.PUT_GUILD_INCIDENT_ACTIONS.compile(guild=guild)

        body = data_binding.JSONObjectBuilder()
        body.put("invites_disabled_until", invites_disabled_until, conversion=date.isoformat() if date else None)
        body.put("dms_disabled_until", dms_disabled_until, conversion=date.isoformat() if date else None)

        response = await self._request(route, json=body)
        assert isinstance(response, dict)
        return self._entity_factory.deserialize_guild_incidents(response)
```